src/main/java/com/rest/controller/ContactRESTController.java

```diff
@@ -0,0 +1,167 @@
+package com.rest.controller;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Set;
+ 
+import javax.validation.ConstraintViolation;
+import javax.validation.Validator;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.Produces;
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import io.dropwizard.auth.Auth;
+
+import com.rest.dao.contact.ContactDao;
+import com.rest.representations.Contact;
+import com.rest.representations.*;
+import com.rest.filter.*;
+import java.util.Optional;  
+import java.util.List;
+import java.util.Collections;
+
+@Path("/user")
+@Produces(MediaType.APPLICATION_JSON)
+public class ContactRESTController {
+ 
+    private final Validator validator;
+    private final ContactDao contactDao = new ContactDao();
+ 
+    public ContactRESTController(Validator validator) {
+        this.validator = validator;
+    }
+ 
+    @GET
+    @Path("/{uid}/contacts/")
+    @AccessCheck
+    @RolesAllowed({"USER"})
```

Comment:
How are authorizer and authenticator getting invoked?

Reply:
So @RolesAllowed calls Authorizer  https://docs.oracle.com/cd/E19798-01/821-1841/gjgcq/index.html and @auth calls Authenticator

Reply:
@auth triggers Authenticator, source- https://howtodoinjava.com/dropwizard/dropwizard-basic-auth-security-example/
<img width="1440" alt="Screenshot 2021-12-29 at 3 17 21 PM" src="https://user-images.githubusercontent.com/94799850/147649087-1d233ed5-9f91-492a-91e0-542d57aee757.png">